File: src/main/webapp/js/utils/PathUtil.js
//@flow
import StringUtil from "./StringUtil";

/**
 * Utility class to contain static functionalities related to string based paths for files or directories.
 */
export default class PathUtil {

  /**
   *
   * @param path {String}
   */
  static normalizePath(path) {
    if (StringUtil.isNotEmpty(path)) {
      //Todo: check for path regex.
    }
    else {
      return null;
    }
  }

  /**
   * To escape all specicl characters related to a file path by backslash.
   * @param path
   * @returns {*}
   */
  static escapeFilePath(path) {
    if (StringUtil.isNotEmpty(path)) {
      if(path.indexOf('\\') > -1)
        path.replace('\\', '\\\\');
      return path;
    }
    else {
      return null;
    }
  }
}

File: src/main/java/com/jbbassociates/utils/PathUtil.java
package com.jbbassociates.utils;

import java.io.File;

/**
 *
 */
public class PathUtil {

  /**
   *
   * @param filePath {String}
   * @return {String}
   */
  public static String normalizePath(String filePath) {
    if (StringUtil.isNotNullOrEmpty(filePath, true)) {
      //Todo: replace '\\' to '/'
      filePath.replaceAll("\\\\", "/");
    }
    return filePath;
  }

  /**
   *
   * @param filePath {String}
   * @return {boolean}
   */
  public static boolean isValidFilePath(String filePath) {
    return PathUtil.isValidPath(filePath) && FileUtil.isExists(new File(filePath));
  }

  /**
   *
   * @param path {String}
   * @return {boolean}
   */
  public static boolean isValidPath(String path) {
    //Todo: check for path regex.
    //&& filePath.matches("")
    return StringUtil.isNotNullOrEmpty(path, true);
  }

}